src/binding_utils.rs

```diff
@@ -56,6 +56,17 @@ macro_rules! py_object {
             PyObject::from_owned_ptr($py, ptr)
         }
     }};
+    ($_self: ident, $subclass: ident, $py: ident, init_non_self) => {{
+        let initializer = PyClassInitializer::from($_self);
```

Comment:
What you need isn't `let initializer = PyClassInitializer::from(($subclass, $class($_self)));`
And you add `$class: ident` instead of `init_non_self` ?

Reply:
Btw if $subclass is not necessary, you can remove from the first implementation, but combine both

Reply:
I modified the code a bit to simplify the macro and just add a `$class` parameter.